cypress/integration/articleFlows/pinArticle.spec.js

```diff
@@ -0,0 +1,48 @@
+describe('Pin an article', () => {
+  beforeEach(() => {
+    cy.testSetup();
+    cy.fixture('users/adminUser.json').as('user');
+
+    cy.get('@user').then((user) => {
+      cy.loginUser(user).then(() => {
+        cy.createArticle({
+          title: 'Test Article',
+          tags: ['beginner', 'ruby', 'go'],
+          content: `This is a test article's contents.`,
+          published: true,
+        }).then((response) => {
+          cy.visit(response.body.current_state_path);
+        });
+      });
+    });
+  });
+
+  it('should pin a post', () => {
+    cy.findByRole('main').within(() => {
+      cy.findByRole('button', { name: 'Pin Post' }).click();
+
+      // check the button has changed to "Unpin Post"
+      cy.findByRole('button', { name: 'Unpin Post' });
+    });
+
+    cy.visit('/');
+
+    cy.findByRole('main').within(() => {
+      cy.findByTestId('pinned-article').should('be.visible');
+    });
```

Comment:
```suggestion
    cy.findByRole('main').findByTestId('pinned-article').should('be.visible');
```

since there's only one action on the `main` we could just chain it